[Getting started][Feature flags] Add "add your code" page (#6502)
PostHog/posthog.com: contents/docs/integrate/feature-flags-code/_snippets/feature-flags-code-api.mdx

## Step 1: Evaluate the feature flag value using the `/decide`

`/decide` is the endpoint used to determine if a given flag is enabled for a certain user or not.

#### Request

```shell
curl -v -L --header "Content-Type: application/json" -d '  {
    "api_key": "<ph_project_api_key>",
    "distinct_id": "distinct_id_of_your_user",

>     "groups" : { # Optional

Reviewer: groups are required when you have any group based flags, those wouldn't work without this

        "group_type": "group_id" # Replace "group_type" with the name of your group type. Replace "group_id" with the id of your group.
    },
    "person_properties": {"<personProp1>": "<personVal1>"}, # Optional. Include any properties used to calculate the value of the feature flag.
    "group_properties": {"group type": {"<groupProp1>":"<groupVal1>"}} # Optional. Include any properties used to calculate the value of the feature flag.

> }' https://app.posthog.com/decide?v=3

Reviewer 2: ```suggestion
}' https://app.posthog.com/decide?v=3 # or https://eu.posthog.com/decide?v=3
```

```

> **Note:** `person_properties` and `group_properties` are optional. 
> 

> > By default, PostHog will use the user and group properties stored in PostHog. You only need to provide `person_properties` and `group_properties` if you wish to use different values than the ones in PostHog.

Reviewer 2: ```suggestion
> By default, flag evaluation uses the user and group properties stored in PostHog. You only need to provide `person_properties` and `group_properties` if you wish to use different values than the ones in PostHog.
```

#### Response

```shell
{
  "config": {
    "enable_collect_everything": true
  },
  "editorParams": {},
  "errorComputingFlags": false,
  "isAuthenticated": false,
  "supportedCompression": [
    "gzip",
    "lz64"
  ],
  "featureFlags": {
    "my-awesome-flag": true,
    "my-awesome-flag-2": true,
    "my-multivariate-flag": "some-string-value",
    "flag-thats-not-on": false,
  },
  "featureFlagPayloads": {
    "my-awesome-flag": "example-payload-string",
    "my-awesome-flag-2": {"color": "blue", "animal": "hedgehog"}
  }
}
```

> > **Note:** `errorComputingFlags` will return `true` if we didn't manage to compute some flags (For example, if there's an [ongoing incident](https://status.posthog.com/)). 

Reviewer 2: ```suggestion
> **Note:** `errorComputingFlags` will return `true` if we didn't manage to compute some flags (for example, if there's an [ongoing incident involving flag evaluation](https://status.posthog.com/)). 
```

> 

> > This allows for making partial updates to currently active flags in your clients.

Reviewer 2: ```suggestion
> This enables partial updates to currently active flags in your clients.
```

> import IncludePropertyInEvents from "./include-feature-flag-property-in-backend-events.mdx" 

Reviewer: also worth noting that you don't have to do this manually in backend SDK always. There's an option to `send_feature_flags_with_event` in most SDKs. The downside here though is that this makes an additional request to `/decide` to get feature flag values, which can slow things down on your backend, unlike doing this manually where ideally you're already using local evaluation to compute flags.

Author: Thanks @neilkakkar . Is the default value for `send_feature_flags_with_event` false?

Which way do we recommend more?

Reviewer: default false yes, recommendation depends on your use case, if you don't care about the extra API call & your server is long-lived, definitely just use the param.

In all other cases (which is probably most people's backend, do it manually)

<IncludePropertyInEvents />

```shell
curl -v -L --header "Content-Type: application/json" -d '  {
    "api_key": "<ph_project_api_key>",
    "distinct_id": "distinct_id_of_your_user",
    "properties": {
        "$feature/feature-flag-key": "variant-key", # replace feature-flag-key with your flag key. Replace 'variant-key' with the key of your variant
    },
    "event": "your_event_name"

> }' https://app.posthog.com/capture/

Reviewer 2: ```suggestion
}' https://app.posthog.com/capture/ # or https://eu.posthog.com/capture/
```

> }' https://app.posthog.com/capture/

Reviewer 2: Should this section go under Step 3? Seems a bit out of place.

```

### Step 3: Send a `$feature_flag_called` event

To track usage of your feature flag and view related analytics in PostHog, submit the `$feature_flag_called` event whenever you check a feature flag value in your code.

You need to include two properties with this event:

> 1. `$feature_flag_response`: this is the name of the variant the user has been assigned to e.g., "control" or "test"

Reviewer 2: ```suggestion
1. `$feature_flag_response`: This is the name of the variant the user has been assigned to e.g., "control" or "test"
```

2. `$feature_flag`: This is the key of the feature flag in your experiment.

```shell
curl -v -L --header "Content-Type: application/json" -d '  {
    "api_key": "<ph_project_api_key>",
    "distinct_id": "distinct_id_of_your_user",
    "properties": {

>         "$feature_flag_response": "variant-name",
>         "$feature_flag": "feature-flag-key"

Reviewer 2: ```suggestion
        "$feature_flag": "feature-flag-key",
        "$feature_flag_response": "variant-name"
```

    },
   "event": "$feature_flag_called"

> }' https://app.posthog.com/capture/

Reviewer 2: ```suggestion
}' https://app.posthog.com/capture/ # or https://eu.posthog.com/capture/
```
